pkg/logger/logger_pipeline.go

```diff
@@ -71,43 +68,57 @@ func (s *SplitPipeline) Write(messages []ChannelMessage) {
 		return
 	}
 
-	finalOutput := make([][]string, s.config.LoggerGroupCount)
+	finalOutput := make([][]string, s.config.GroupContainer.GroupCount())
 
 	for _, m := range messages {
-		if len(s.config.LoggerGroup) < m.Logger.ID() {
+		loggerGroup := s.config.GroupContainer.GroupByLogger(m.Logger)
+		if loggerGroup == nil {
 			fmt.Fprintln(s.writer, "Received a logger with the id ", m.Logger.ID(), "that could not be sorted")
-			continue
 		}
 
-		loggerGroupID := s.config.LoggerGroup[m.Logger.ID()]
-
 		message := m.MessageAsString()
+
+		var slices []string
 		if s.config.ShowLoggerName {
-			message = "[" + m.Logger.Name() + "] " + message
+			// Slice it but with prefix
+			slices = ChunkSlice(message, s.config.CharacterPerPipe-loggerGroup.MaxPrefixSize())
+
+			loggerPrefix := m.Logger.AsPrefix()
+			for index, slice := range slices {
+				slices[index] = loggerPrefix + slice
+			}
+		} else {
+			slices = ChunkSlice(message, s.config.CharacterPerPipe) // slice it normally
 		}
 
-		slices := ChunkSlice(message, s.config.CharacterPerPipe)
-		loggerGroup := finalOutput[loggerGroupID]
+		loggerGroupOutput := finalOutput[loggerGroup.ID()] // Take output till now
 
-		loggerGroup = append(loggerGroup, slices...)
-		finalOutput[loggerGroupID] = loggerGroup
+		loggerGroupOutput = append(loggerGroupOutput, slices...) // append new output
+		finalOutput[loggerGroup.ID()] = loggerGroupOutput        // set output
 	}
 
 	for line := 0; true; line++ {
 
 		output := ""
 		empty := true
 
-		for group := 0; group < s.config.LoggerGroupCount; group++ {
-
+		for group := 0; group < s.config.GroupContainer.GroupCount(); group++ {
 			groupArray := finalOutput[group]
+			groupArraySize := len(groupArray)
 			if group > 0 { // Print new line if we are not first or last message
 				output += PipelineSeparator
 			}
 
-			if line < len(groupArray) {
+			if line < groupArraySize {
 				groupOutput := groupArray[line]
-				output += groupOutput + strings.Repeat(" ", s.config.CharacterPerPipe-bunt.PlainTextLength(groupOutput)) + "\u001b[0m"
+				output += groupOutput
+
+				if s.config.GroupContainer.GroupCount() > 1 {
+					emptyChars := s.config.CharacterPerPipe - bunt.PlainTextLength(groupOutput) // The prefix is already in the group output
+					output += strings.Repeat(" ", emptyChars)
```

Comment:
this will eventually give you a negative number, check this.

Reply:
@LynxPlay101 I did not have time to review this one here. If the other work scales down, I will have a look. Until then, what do you think about the comment made by Enrique?

Reply:
Closing it was my response 😂 The new branch contains a first test push, and works fine. I could not reproduce the negative repeating numbers